internal/storage/v2/clickhouse/tracestore/dbmodel/to_dbmodel.go

```diff
@@ -1,550 +0,0 @@
-
-				var linksTraceId []string
-				var linksSpanId []string
-				var linksTracesState []string
-				var linkNestedGroup NestedAttributesGroup
-				for _, link := range span.Links().All() {
-					linksTraceId = append(linksTraceId, traceIDToHexString(link.TraceID()))
-					linksSpanId = append(linksSpanId, spanIDToHexString(link.SpanID()))
-					linksTracesState = append(linksTracesState, link.TraceState().AsRaw())
-					linkGroup := attributesToGroup(link.Attributes())
-					linkNestedGroup.AttributesGroups = append(linkNestedGroup.AttributesGroups, linkGroup)
-				}
-				linksSpanIdCol := traceColumnSet.links.spanID.Col
-				linksSpanIdCol.(*proto.ColArr[string]).Append(linksSpanId)
-				linksTraceIdCol := traceColumnSet.links.traceID.Col
-				linksTraceIdCol.(*proto.ColArr[string]).Append(linksTraceId)
-				linksTraceStateCol := traceColumnSet.links.traceState.Col
-				linksTraceStateCol.(*proto.ColArr[string]).Append(linksTracesState)
-
-				traceColumnSet.resource.attributes.appendAttributeGroup(resourceGroup)
-				traceColumnSet.scope.attributes.appendAttributeGroup(scopeGroup)
-				traceColumnSet.span.attributes.appendAttributeGroup(spanGroup)
-				traceColumnSet.events.attributes.appendNestedAttributeGroup(eventNestedGroup)
-				traceColumnSet.links.attributes.appendNestedAttributeGroup(linkNestedGroup)
-			}
-		}
-	}
-
-	input := proto.Input{}
-	input = append(input, traceColumnSet.span.spanInput()...)
-	input = append(input, traceColumnSet.scope.scopeInput()...)
-	input = append(input, traceColumnSet.resource.resourceInput()...)
-	input = append(input, traceColumnSet.events.eventsInput()...)
-	input = append(input, traceColumnSet.links.linkInput()...)
-	return input
-}
-
-// NestedAttributesGroup There is a one-to-many relationship between a NestedAttributesGroup and a pcommon.Map.
-// ptrace.SpanEventSlice and ptrace.SpanLinkSlice are stored in a Nested format in the database.
-// Since all arrays in Nested need to have the same length, AttributesGroup cannot be used directly.
-type NestedAttributesGroup struct {
-	AttributesGroups []AttributesGroup
-}
-
-// AttributesGroup captures all data from a single pcommon.Map, except
-// complex attributes (like slice or map) which are currently not supported.
-// AttributesGroup consists of pairs of vectors for each of the supported primitive
-// types, e.g. (BoolKeys, BoolValues). Every attribute in the pcommon.Map is mapped
-// to one of the pairs depending on its type. The slices in each pair have identical
-// length, which may be different from length in another pair. For example, if the
-// pcommon.Map has no Boolean attributes then (BoolKeys=[], BoolValues=[]).
-type AttributesGroup struct {
-	BoolKeys     []string
-	BoolValues   []bool
-	DoubleKeys   []string
-	DoubleValues []float64
-	IntKeys      []string
-	IntValues    []int64
-	StrKeys      []string
-	StrValues    []string
-	BytesKeys    []string
-	BytesValues  []string
-}
```

Comment:
@yurishkuro I don't believe this approach for attributes is compatible with ClickHouse. If we want to maintain typed attributes, I think using the [JSON](https://clickhouse.com/docs/sql-reference/data-types/newjson) type might be the best approach here. Thoughts?